integration-tests/test/test_slash.py

```diff
@@ -35,39 +40,83 @@ def hex_block_hash(block_hash: bytes) -> bytes:
+    with three_nodes_network_with_node_client(command_line_options, random_generator, docker_client) as  (context, _ , validator1, validator2, client):
+        contract = '/opt/docker/examples/tut-hello.rho'
+
+        validator1.deploy(contract, BONDED_VALIDATOR_KEY_1)
+        blockhash = validator1.propose()
 
-            wait_for_node_sees_block(context, validator2, blockhash)
+        wait_for_node_sees_block(context, validator2, blockhash)
 
-            block_info = validator1.show_block_parsed(blockhash)
+        block_info = validator1.show_block_parsed(blockhash)
 
-            block_msg = client.block_request(block_info['blockHash'], validator1)
-            evil_block_hash = generate_block_hash()
+        block_msg = client.block_request(block_info['blockHash'], validator1)
+        evil_block_hash = generate_block_hash()
 
-            block_msg.blockHash = evil_block_hash
-            block_msg.sig = BONDED_VALIDATOR_KEY_1.sign_block_hash(evil_block_hash)
+        block_msg.blockHash = evil_block_hash
+        block_msg.sig = BONDED_VALIDATOR_KEY_1.sign_block_hash(evil_block_hash)
 
-            client.send_block(block_msg, validator2)
+        client.send_block(block_msg, validator2)
 
-            record_invalid = re.compile("Recording invalid block {}... for InvalidBlockHash".format(hex_block_hash(evil_block_hash)[:10].decode('utf8')))
-            wait_for_log_match(context, validator2, record_invalid)
+        record_invalid = re.compile("Recording invalid block {}... for InvalidBlockHash".format(hex_block_hash(evil_block_hash)[:10].decode('utf8')))
+        wait_for_log_match(context, validator2, record_invalid)
 
-            validator2.deploy(contract, BONDED_VALIDATOR_KEY_2)
+        validator2.deploy(contract, BONDED_VALIDATOR_KEY_2)
 
-            slashed_block_hash = validator2.propose()
+        slashed_block_hash = validator2.propose()
 
-            block_info = validator2.show_block_parsed(slashed_block_hash)
-            bonds_validators = extract_validator_stake_from_bonds_validator_str(block_info['bondsValidatorList'])
+        block_info = validator2.show_block_parsed(slashed_block_hash)
+        bonds_validators = extract_validator_stake_from_bonds_validator_str(block_info['bondsValidatorList'])
 
-            assert bonds_validators[BONDED_VALIDATOR_KEY_1.get_public_key().to_hex()] == 0
+        assert bonds_validators[BONDED_VALIDATOR_KEY_1.get_public_key().to_hex()] == 0
+
+@pytest.mark.skipif(sys.platform in ('win32', 'cygwin', 'darwin'), reason="Only Linux docker support connection between host and container which node client needs")
+def test_slash_invalid_block_seq(command_line_options: CommandLineOptions, random_generator: Random, docker_client: DockerClient) -> None:
+    """
+    Propose an block with invalid block number(a block number that isn't one more than the max of all the parents block's numbers).
```

Comment:
Block number != block sequence number

Block sequence number is basically a per validator block number.